ppa/archive/management/commands/hathi_images.py

```diff
@@ -0,0 +1,286 @@
+"""
+**hathi_images** is a custom manage command for downloading both full-size
+and thumbnail page images for a list of HathiTrust volumes.
+"""
+import argparse
+from collections import Counter
+from collections.abc import Iterable
+import logging
+import requests
+from pathlib import Path
+import signal
+import time
+from typing import Self
+
+from tqdm import tqdm
+from django.core.management.base import BaseCommand, CommandError
+from django.template.defaultfilters import pluralize
+from corppa.utils.path_utils import encode_htid, get_vol_dir
+
+from ppa.archive.models import DigitizedWork
+from ppa.archive.templatetags.ppa_tags import page_image_url
+
+logger = logging.getLogger(__name__)
+
+
+class DownloadStats:
```

Comment:
this seems complicated but I see the value of it for wrangling multiple counts and displaying them nicely

Reply:
FWIW, this is pretty much just a wrapped tuple of Counters with a pretty print.